pkg/translate/translate.go

```diff
@@ -389,12 +392,48 @@ func (t *Translator) OverlayK8sSettings(yml string, icp *v1alpha2.IstioControlPl
 		if err != nil {
 			return "", fmt.Errorf("could not marshal to YAML in OverlayK8sSettings: %s", err)
 		}
-		mergedYAML, err := overlayK8s(baseYAML, overlayYAML, path[1:])
+
+		var dstNodeYaml, k8sSettingName string
+		if len(path) != 0 {
+			k8sSettingName = path[len(path)-1]
+		}
+
+		switch k8sSettingName {
```

Comment:
Suggest moving all this code into overlayK8s, otherwise the function will be too long and complex to understand.

Reply:
done.